src/controllers/DynamicFacetQueryController.ts

```diff
@@ -54,11 +63,24 @@ export class DynamicFacetQueryController {
       freezeCurrentValues: this.freezeCurrentValues,
       isFieldExpanded: this.numberOfValuesToRequest > this.facet.options.numberOfValues
     };
+  }
 
-    queryBuilder.facetRequests.push(facetRequest);
-    if (this.freezeFacetOrder) {
-      queryBuilder.facetOptions.freezeFacetOrder = true;
+  public executeIsolatedQuery(): Promise<IQueryResults> {
+    const query = this.facet.queryController.getLastQuery();
+    // @ts-ignore
+    query.isInternalQuery = true;
+    query.numberOfResults = 0;
+
+    const previousFacetRequestIndex = findIndex(query.facets, { facetId: this.facet.options.id });
+    if (previousFacetRequestIndex !== -1) {
+      query.facets[previousFacetRequestIndex] = this.facetRequest;
+    } else if (query.facets) {
+      query.facets.push(this.facetRequest);
+    } else {
+      query.facets = [this.facetRequest];
     }
+
+    return this.facet.queryController.getEndpoint().search(query);
```

Comment:
Are errors in the request handled by the call to `.search`? If not, we should have a `catch` and return an appropriate value. I would suggest an "empty" `IQueryResults` object.

Reply:
don't need to return a value, it would just make the entire values disappear, we will just cancel the loading animation.